From 8b4c37628831828cd651ec6820121224586f6ed2 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Wed, 13 Feb 2013 12:47:12 +0100
Subject: BUILD: fix a warning emitted by isblank() on non-c99 compilers

Commit a2b9dad introduced use of isblank() which is not present everywhere
and requires -std=c99 or various other defines. Here on gcc-3.4 with glibc
2.3 it emits a warning though it works :

  src/checks.c: In function 'event_srv_chk_r':
  src/checks.c:1007: warning: implicit declaration of function 'isblank'

This macro matches only 2 values, better replace it with the explicit match.
---
 src/checks.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/checks.c b/src/checks.c
index 1a53669..2ab509d 100644
--- a/src/checks.c
+++ b/src/checks.c
@@ -1004,10 +1004,10 @@ static void event_srv_chk_r(struct connection *conn)
 			 * The command keyword must terminated the string or
 			 * be followed by a blank.
 			 */
-			if (end[0] == '\0' || isblank(end[0])) {
+			if (end[0] == '\0' || end[0] == ' ' || end[0] == '\t') {
 				status = HCHK_STATUS_L7STS;
 				/* Skip over leading blanks */
-				while (end[0] != '\0' && isblank(end[0]))
+				while (end[0] != '\0' && (end[0] == ' ' || end[0] == '\t'))
 					end++;
 				desc = end;
 			}
-- 
1.7.1

